kbase-extension/static/kbase/js/common/jobs.js

```diff
@@ -1,37 +1,129 @@
+        span = t('span');
+
+    const cssBaseClass = 'kb-job-status',
         jobNotFound = [
             'This job was not found, or may not have been registered with this Narrative.',
-            'You will not be able to inspect the job status or view the job log'
+            'You will not be able to inspect the job status or view the job log',
         ],
-        jobStatusUnknown = ['Determining job state...'];
+        jobStatusUnknown = ['Determining job state...'],
+        // valid job states from ee2, plus 'does_not_exist'
+        validJobStates = [
+            'created',
+            'estimating',
+            'queued',
+            'running',
+            'completed',
+            'error',
+            'terminated',
+            'does_not_exist',
+        ];
+
+    /**
+     * A jobState object is deemed valid if
+     * 1. It's an object (not an array or atomic type)
+     * 2. It has a 'created' key
+     * 3. It has a 'job_id' key
+     * 4. It has a 'status' key, which appears in the array validJobStates
+     *
+     * There are other required fields, but these checks are sufficient for the UI.
+     *
+     * See execution_engine2/lib/execution_engine2/db/models/models.py in the
+     * execution_engine2 repo for the full job spec.
+     *
+     * This function should be updated to stay in sync with ee2's output.
+     *
+     * @param {object} jobState
+     * @returns {boolean} true|false
+     */
+    function isValidJobStateObject(jobState) {
+        const requiredProperties = ['job_id', 'created', 'status'];
+        return (
+            jobState !== null &&
+            typeof jobState === 'object' &&
+            requiredProperties.every((prop) => prop in jobState) &&
+            validJobStates.includes(jobState.status)
+        );
+    }
+
+    /**
+     * Given a job status string, ensure that it is valid
+     *
+     * @param {*} status
+     * @returns {boolean} true|false
+     */
+    function isValidJobStatus(status) {
+        return status && validJobStates.includes(status);
+    }
+
+    /**
+     * A jobInfo object should have the minimal structure
+     *
+     * {
+     *      job_id: ...
+     *      job_params: [
+     *          {
+     *              param_one: 'value one',
+     *              param_two: 'value two',
+     *              ...
+     *          }
+     *      ]
+     * }
+     *
+     * This function should be updated to stay in sync with ee2's output.
+     *
+     * @param {object} jobInfo
+     * @returns {boolean} true|false
+     */
+    function isValidJobInfoObject(jobInfo) {
+        try {
+            if (
+                'job_id' in jobInfo &&
+                Object.prototype.toString.call(jobInfo.job_params[0]) === '[object Object]' &&
+                Object.keys(jobInfo.job_params[0]).length > 0
+            ) {
+                return true;
+            }
+        // eslint-disable-next-line no-empty
+        } catch (err) {}
```

Comment:
we're just checking whether the job info object is valid; we don't really care how it deviates from the expected structure, so this error can be ignored